docs/development/codeguide_emacs.rst

```diff
@@ -1,207 +1,67 @@
-===========================================
-
-The "fill-column" column is highlighted in red. For this to work,
-download `column-marker.el
-<http://www.emacswiki.org/emacs/column-marker.el>`_ and place it in the
-Emacs configuration directory.
-
-.. code-block:: scheme
-
-  ;; Highlight character at "fill-column" position.
-  (require 'column-marker)
-  (set-face-background 'column-marker-1 "red")
+  ;; Automatically remove trailing whitespace when file is saved.
   (add-hook 'python-mode-hook
-            (lambda () (interactive)
-              (column-marker-1 fill-column)))
+    (lambda () (add-to-list 'write-file-functions 'delete-trailing-whitespace)))
 
-Flymake
-=======
-
-Flymake will mark lines that do not satisfy syntax requirements in
-red. When cursor is on such a line a message is displayed in the
-mini-buffer. When mouse pointer is on such a line a "tool tip" message
-is also shown.
-
-For flymake to work with `pycodestyle`_ and `pyflakes`_, create an
-executable file named `pychecker`_ with the following contents. This
-file must be in the system path.
-
-.. code-block:: sh
-
-  #!/bin/bash
-
-  pyflakes "$1"
-  pycodestyle --ignore=E221,E701,E202 --repeat "$1"
-  true
-
-Also download `flymake-cursor.el
-<http://www.emacswiki.org/emacs/flymake-cursor.el>`_ and place it in the
-Emacs configuration directory.  Then add the following code to the Emacs
-configuration:
-
-.. code-block:: scheme
-
-  ;; Setup for Flymake code checking.
-  (require 'flymake)
-  (load-library "flymake-cursor")
-
-  ;; Script that flymake uses to check code. This script must be
-  ;; present in the system path.
-  (setq pycodechecker "pychecker")
+If you want to use this for every type of file, you can use
+``(add-hook 'before-save-hook 'delete-trailing-whitespace)``.
 
-  (when (load "flymake" t)
-    (defun flymake-pycodecheck-init ()
-      (let* ((temp-file (flymake-init-create-temp-buffer-copy
-                         'flymake-create-temp-inplace))
-             (local-file (file-relative-name
-                          temp-file
-                          (file-name-directory buffer-file-name))))
-        (list pycodechecker (list local-file))))
-    (add-to-list 'flymake-allowed-file-name-masks
-                 '("\\.py\\'" flymake-pycodecheck-init)))
+Flycheck
+========
 
-  (add-hook 'python-mode-hook 'flymake-mode)
-
-.. note::
-
-    Flymake will save files with suffix *_flymake* in the current
-    directory. If it crashes for some reason, then these files will not
-    get deleted.
-
-    Sometimes there is a delay in refreshing the results.
-
-Delete trailing white spaces and blank lines
-============================================
-
-To manually delete trailing whitespaces, press ``C-t C-w``, which will run
-the command "delete-whitespaces`. This command is also run when a file is
-saved, and hence all trailing whitespaces will be deleted on saving a Python
-file.
-
-To make sure that all "words" are separated by only one space, type
```

Comment:
This only works for space between two words that the cursor is on. More work to remember the key combination than to just do it...